manifests/pkcs11_eventmgr.pp

```diff
@@ -64,7 +75,11 @@
     $pkcs11_module_file = $pkcs11_module
   }
 
-  $merged_event_opts = merge($event_opts_base, $event_opts)
+  if $lock_screen_on_card_removal {
+    $merged_event_opts = $event_opts_base + $event_opts_screen_lock + $event_opts
```

Comment:
Here the `$event_opts` potentially overwrite the `$event_opts_screen_lock`. Is that intentional? For me the other way around would be more intuitive.

In any case, I suggest to define this in the class "docstring".

Reply:
I guess this can go either way...

I thought of `event_opts` as some kind of overwrite that is more urgent than the 'syntax sugar' that is implemented with the `lock_screen_on_card_removal` flag.

There isn't any parameter documented in this class, so I adapted that, but you are right, that this and all other parameters should be documented.

Reply:
Hmm... It does make sense to give the client the "last word" to overwrite any default or generated config. This would suggest the current version you have.

Reply:
I added parameter documentation now. I also found some more type and formatting issues, which I improved as well.